### PR TITLE
Refactor Invoice and InvoiceRepo classes to pass Spec Harness

### LINKED ISSUE #97: Refactor Invoice & InvoiceRepo for Spec Harness Compliance
We need to run the spec harness on the Invoice and Invoice repo files and make all necessary updates.

### PR DESCRIPTION
**Pull request by mkrumholz**
## Changes

Updated formatting for Invoice attributes to match the spec harness requirements. 

## Requires Review

SalesEngine updates from #88 have been merged and this branch has no conflicts there. We should be ready to merge. 

## Checklist
- [x] Tests pass
- [x] Ready to merge?

closes #97

### CODE REVIEW

File: lib/invoice.rb [hunk 1]
```diff
@@ -2,15 +2,13 @@ class Invoice
   attr_reader :id,
               :customer_id,
               :merchant_id,
-              :status,
-              :created_at,
-              :updated_at
```
Comment (NoahZinter) [inline lib/invoice.rb:7]: Should we be dropping time attr_readers for the other classes too, for spec harness?

File: lib/invoice.rb [hunk 2]
```diff
@@ -2,15 +2,13 @@ class Invoice
...
+              :status
 
   def initialize(transaction_details)
-    @id = transaction_details[:id]
-    @customer_id = transaction_details[:customer_id]
-    @merchant_id = transaction_details[:merchant_id]
-    @status = transaction_details[:status]
+    @id = transaction_details[:id].to_i
+    @customer_id = transaction_details[:customer_id].to_i
+    @merchant_id = transaction_details[:merchant_id].to_i
```
Comment (NoahZinter) [inline lib/invoice.rb:10]: Nice

File: lib/invoice.rb [hunk 3]
```diff
@@ -26,4 +24,16 @@ def update_status(status)
   def update_time
     @updated_at = Time.now
   end
+
+  def created_at
+    return @created_at if @created_at.instance_of?(Time)
```
Comment (NoahZinter) [inline lib/invoice.rb:29]: Are these escape clauses for rspec?

File: lib/sales_analyst.rb [hunk 4]
```diff
@@ -60,23 +60,23 @@ def average_item_price_for_merchant(merchant_id)
 
   def average_average_price_per_merchant
     sum_of_averages = all_merchants.sum do |merchant|
-      average_item_price_for_merchant(merchant.id) unless num_of_items_per_merchant[merchant] == 0
+      average_item_price_for_merchant(merchant.id) unless num_of_items_per_merchant[merchant].zero?
```
Comment (NoahZinter) [inline lib/sales_analyst.rb]: oh cool, didn't know .zero? was a thing

File: spec/sales_engine_spec.rb [hunk 5]
```diff
@@ -1,22 +1,43 @@
...
+  data_hashes = []
+  10.times { data_hashes << details }
+
   describe '#from_csv' do
     it 'creates a new instance of SalesEngine' do
       mock_items = ItemMocks.items_as_mocks(self)
       allow_any_instance_of(ItemRepository).to receive(:create_items).and_return(mock_items)
       mock_merchants = MerchantMocks.merchants_as_mocks(self)
       allow_any_instance_of(MerchantRepository).to receive(:create_merchants).and_return(mock_merchants)
-      files = { items: './file1.csv', merchants: './file2.csv' }
```
Comment (NoahZinter) [inline spec/sales_engine_spec.rb]: Are we going to run into merge conflicts between this and pull #88? They wouldn't be hard to resolve but we're definitely touching the same file here

**Review by NoahZinter — APPROVED** [review]
Looks solid! I think we'll have to resolve some merge issues between this and the content of pull#88 since we're touching sales engine in both?

File: lib/invoice.rb (hunk 1 above)
Comment (mkrumholz) [reply lib/invoice.rb:7]: Yes! We will need to do that. Good catch! Will update.

Comment (mkrumholz) [reply lib/invoice.rb:7]: @NoahZinter yea, check out my Wiki on the spec harness. Need to allow for different inputs, so the attr_readers are duplicative

File: lib/invoice.rb (hunk 3 above)
Comment (mkrumholz) [reply lib/invoice.rb:29]: @NoahZinter for the spec harness -- they give us inputs in both Time and String formats, so it has to deal with either one coming in but always return a Time object

File: lib/sales_analyst.rb (hunk 4 above)
Comment (mkrumholz) [reply lib/sales_analyst.rb]: Tbh this must have been a change from @redferret maybe? I don't recall doing this and/or knowing that existed either 🤔

File: spec/sales_engine_spec.rb (hunk 5 above)
Comment (mkrumholz) [reply spec/sales_engine_spec.rb]: I just merged this branch into that one @NoahZinter so as long as we merge this one first, it'll be perfectly fine. I'll resolve any issues in there before we merge that in (don't think there should be any though).

**Comment by mkrumholz** [comment]
@NoahZinter I've got this branch on my local so I'm going to merge the branch for #88 into this one and deal with any conflicts on my local before we merge it.

File: spec/file_io_spec.rb [hunk 6]
```diff
@@ -1,5 +1,5 @@
 require 'rspec'
-require 'CSV'
+require 'csv'
```
Comment (sami-p) [inline spec/file_io_spec.rb:2]: Great catch!

**Review by sami-p — APPROVED** [review]
Perfect!

**Pull request merged**